content/en/docs/components/pipelines/v2/compile-a-pipeline.md

```diff
@@ -1,104 +1,75 @@
+| `type_check` | `bool` | _Optional_<br/>Indicates whether static type checking is enabled during compilation.<br/>For more information about type checking, see [Component I/O: Component interfaces and type checking][type-checking].
 
 ## IR YAML
-When you compile a pipeline it is written to intermediate representation (IR) YAML. An IR YAML is an instance of the [`PipelineSpec`][pipeline-spec] protocol buffer message type, a platform-agnostic pipeline representation protocol.
-
-IR YAML is considered an intermediate representation because the KFP backend compiles `PipelineSpec` to [Argo Workflow][argo-workflow] YAML as the final execution definition for execution on Kubernetes.
-
-Unlike v1 component YAML, IR YAML is not intended to be written directly. For a KFP v2 authoring experience similar to the v1 component YAML authoring experience, see [Author a Pipeline: Custom Container Components][custom-container-component-authoring].
-
-IR YAML contains 7 top-level fields:
-
-**components**
-
-The [`components`][components-schema] section is a map of component name to [`ComponentSpec`][component-spec] for all components used in the pipeline. `ComponentSpec` defines the interface (inputs and outputs) of a component. For primitive components, `ComponentSpec` contains a reference to the executor containing the component implementation. For pipelines used as components, `ComponentSpec` contains a [DagSpec][dag-spec] which includes references to its underlying primitive components.
-
-[Example][components-example]
-
-**deployment_spec**
-
-The [`deployment_spec`][deployment-spec-schema] section contains a map of executor name to [`ExecutorSpec`][executor-spec]. `ExecutorSpec` contains the implementation for a primitive component.
-
-[Example][deployment-spec-example]
-
-**root**  
-[`root`][root-schema] defines the steps of the outermost (root) pipeline definition. It is itself a [`ComponentSpec`][component-spec]. This is the pipeline executed when the YAML is submitted.
-
-[Example][root-example]
-
-**pipeline_info**  
-[`pipeline_info`][pipeline-info-schema] contains pipeline metadata, including the pipeline name.
-
-[Example][pipeline-info-example]  
-
-**sdk_version**  
-[`sdk_version`][sdk-version-schema] records which version of the KFP SDK compiled the pipeline.
-
-[Example][sdk-version-example]  
 
-**schema_version**  
-[`schema_version`][schema-version-schema] records which version of the `PipelineSpec` schema is used for the IR YAML.
+The IR YAML is an intermediate representation of a compiled pipeline or component. It is an instance of the [`PipelineSpec`][pipeline-spec] protocol buffer message type, which is a platform-agnostic pipeline representation protocol. It is considered an intermediate representation because the KFP backend compiles `PipelineSpec` to [Argo Workflow][argo-workflow] YAML as the final execution definition for execution on Kubernetes.
```

Comment:
nit/not your work (this was my language):

`final execution definition for execution on Kubernetes` -> `final pipeline definition for execution`